### PR TITLE
Add client fetch streams

### PR DESCRIPTION
**Pull request by bpafoshizle**
Hello!

### Background ###
First off, thank you all for working on this project. I've been using version 1 for several months now as part of a personal hobby project, and have gotten much use from your efforts. 

I have implemented the code in this PR as part of upgrading some things on that project and wanted to update to TwitchIO version 2, but found some differences. 

If this is useful, I would love to contribute. It would actually be my first ever official contribution to the open source community I've so greatly benefitted from my whole career. 

I apologize if I've missed the mark as to your intentions with the project and this will not fit with that, or if I've made any other code style, testing, or otherwise contribution etiquette snafus. I've tried to follow the patterns and organization I've noticed. Like I said, this is my first time doing this, so the experience itself has been the reward for me so far. If you find this useful, that would just be gravy on top. I'm more than happy to take any coaching to bring it up to your standards. 

### Testing ###

Set your secrets as environment variables, e.g.:

```bash
export TWITCH_BOT_CLIENT_ID=my_id
export TWITCH_BOT_CLIENT_SECRET=s3cr3t
```

Imports, client setup, and run the method:

```python
import os
import asyncio
import twitchio

twitch_client = twitchio.Client.from_client_credentials(
            client_id=os.getenv("TWITCH_BOT_CLIENT_ID"),
            client_secret=os.getenv("TWITCH_BOT_CLIENT_SECRET"),
        )

# any random live channel on twitch
check_channels=["sykkuno"]

streams = asyncio.run(twitch_client.fetch_streams(user_logins=check_channels))
```

### Notes: ###
* I saw that that `get_streams` was already implemented in the http.py module, so I just build the client wrapper
* I had an issue calling `Client.from_client_credentials` so I made my best guess tweak to add the `heartbeat` parameter similar to how was done in `__init__`
* I added a Stream class to the models.py just following the patterns I could see for the other classes in that module
* I added a login property to the PartialUser class, since the get_streams allows you to search with login, and it also returns it in the data. Seemed to make sense.

### CODE REVIEW

File: twitchio/models.py [hunk 1]
```diff
@@ -530,3 +530,38 @@ def __init__(self, data: dict):
...
+        "title",
+        "viewer_count",
+        "started_at",
+        "language",
+        "thumbnail_url",
+        "tag_ids",
+        "is_mature",
+    )
+
+    def __init__(self, http: "TwitchHTTP", data: dict):
```
Comment (IAmTomahawkx) [inline twitchio/models.py:552]: You've made all of these attributes tuples, and please turn parameters that can be turned into ints into such

File: twitchio/user.py [hunk 2]
```diff
@@ -48,11 +48,12 @@
...
 class PartialUser:
 
-    __slots__ = "id", "name", "_http", "_cached_rewards"
+    __slots__ = "id", "name", "login", "_http", "_cached_rewards"
 
-    def __init__(self, http: "TwitchHTTP", id: Union[int, str], name: Optional[str]):
+    def __init__(self, http: "TwitchHTTP", id: Union[int, str], name: Optional[str], login: Optional[str]):
         self.id = int(id)
         self.name = name
+        self.login = login
```
Comment (IAmTomahawkx) [inline twitchio/user.py:56]: I actually purposefully ignored the login because name.lower() is almost always equal to the login
(I've seen one edge case with some funky characters), and a good chunk of the api just straight up doesn't return the login key, so it doesn't make sense to implement

File: twitchio/user.py [hunk 3]
```diff
@@ -621,7 +622,7 @@ def __init__(self, http: "TwitchHTTP", data: dict):
         self.title: str = data["title"]
         self.thumbnail_url: str = data["thumbnail_url"]
         self.live: bool = data["is_live"]
-        self.started_at = datetime.datetime.strptime(data["expires_at"], "%Y-%m-%dT%H:%M:%SZ") if self.live else None
+        self.started_at = datetime.datetime.strptime(data["started_at"], "%Y-%m-%dT%H:%M:%SZ") if self.live else None
         self.tag_ids: List[str] = data["tag_ids"]
```
Comment (IAmTomahawkx) [inline twitchio/user.py:626]: Ideally this would best be done in its own pull request, as it's not relevant to this one

File: twitchio/client.py [hunk 4]
```diff
@@ -480,6 +486,46 @@ async def fetch_tags(self, ids: List[str] = None):
         data = await self._http.get_stream_tags(ids)
         return [models.Tag(x) for x in data]
 
+    async def fetch_streams(
+        self,
+        user_ids: List[str] = None,
```
Comment (IAmTomahawkx) [inline twitchio/client.py:491]: user_ids should be List[int], not List[str]

File: twitchio/client.py [hunk 5]
```diff
@@ -120,6 +125,7 @@ def from_client_credentials(
         self = cls.__new__(cls)
         self.loop = loop or asyncio.get_event_loop()
         self._http = TwitchHTTP(self, client_id=client_id, client_secret=client_secret)
+        self._heartbeat = heartbeat
```
Comment (IAmTomahawkx) [inline twitchio/client.py:128]: Ideally this would best be done in its own pull request, as it's not relevant to this one

**Review by IAmTomahawkx — CHANGES_REQUESTED** [review]
Thanks for the pr, there's a few things that need some fixing up but other than that it looks fine, thanks!

File: twitchio/models.py (hunk 1 above)
Comment (bpafoshizle) [reply twitchio/models.py:552]: Strange they all became tuples. I wonder if black formatter did that at one point. Should be correct now. Thanks for the feedback!

File: twitchio/client.py (hunk 4 above)
Comment (bpafoshizle) [reply twitchio/client.py]: Have made the change. Thanks!

File: twitchio/user.py (hunk 2 above)
Comment (bpafoshizle) [reply twitchio/user.py]: Makes sense. I was wondering about that. I've removed the login related code.

File: twitchio/user.py (hunk 3 above)
Comment (bpafoshizle) [reply twitchio/user.py]: I've removed this change. Will follow up with another PR.
